robocorp-code/src/robocorp_code/inspector/image/_image_inspector.py

```diff
@@ -0,0 +1,135 @@
+        threading.Thread.__init__(self)
+
+        self._stop_event = threading.Event()
+        self.bridge = bridge
+        self.on_pick = on_pick
+        self.confidence = confidence
+
+    def run(self) -> None:
+        try:
+            self._run()
+        except Exception:
+            import traceback
+
+            traceback.print_exc()
+
+    def _run(self) -> None:
+        assert self.bridge is not None
+
+        result_dict = self.bridge.pick(confidence=self.confidence)
+        self.on_pick(result_dict)
+
+
+class _ValidateThread(threading.Thread):
+    def __init__(
+        self,
+        bridge: Optional[ImageBridge],
+        image_base64: str,
+        confidence: Optional[int],
+        on_validate,
+    ) -> None:
+        threading.Thread.__init__(self)
+
+        self._stop_event = threading.Event()
+        self.bridge = bridge
+        self.image_base64 = image_base64
+        self.confidence = confidence
+        self.on_validate = on_validate
+
+    def run(self) -> None:
+        try:
+            self._run()
+        except Exception:
+            import traceback
+
+            traceback.print_exc()
+
+    def _run(self) -> None:
+        assert self.bridge is not None
+
+        matches = self.bridge.validate(
+            image_base64=self.image_base64, confidence=self.confidence
+        )
+        self.on_validate(matches)
+
+
+class IOnPickCallback(typing.Protocol):
+    def __call__(self, locator_info_tree: List[dict]):
+        pass
+
+    def register(self, callback: Callable[[Any], Any]) -> None:
+        pass
+
+    def unregister(self, callback: Callable[[Any], Any]) -> None:
+        pass
+
+
+class ImageInspector:
+    def __init__(self, endpoint: Optional[IEndPoint] = None) -> None:
+        """
+        Args:
+            endpoint: If given notifications on the state will be given.
+        """
+        # callbacks
+        self.on_pick: IOnPickCallback = Callback()
+        self.on_validate: IOnPickCallback = Callback()
+        # internals
+        self._image_bridge = ImageBridge(endpoint=endpoint, logger=log)
+        # threads
+        self._picker_thread: Optional[_PickerThread] = None
+        self._validate_thread: Optional[_ValidateThread] = None
+
+    def start_pick(self, confidence: Optional[int] = None) -> None:
+        log.info(">>> Image - Start pick...")
+
+        self._picker_thread = _PickerThread(
+            bridge=self._image_bridge,
+            confidence=confidence,
+            on_pick=self.on_pick,
+        )
+        self._picker_thread.start()
+
+    def stop_pick(self) -> None:
+        log.info(">>> Image - Stop pick...")
+        if self._picker_thread:
+            self._picker_thread._stop_event.set()
+        if self._validate_thread:
+            self._validate_thread._stop_event.set()
+        self._image_bridge.stop()
+
+    def validate(self, image_base64: str, confidence: Optional[int] = None) -> None:
+        log.info(">>> Image - Validate pick...")
```

Comment:
You can leave the validate scaffolding, but we shouldn't call it at this point...